components/net/fetch/methods.rs

```diff
@@ -874,6 +887,37 @@ fn is_null_body_status(status: &HttpStatus) -> bool {
     )
 }
 
+fn should_response_be_blocked_by_csp(
+    request: &Request,
+    response: &Response,
+) -> (bool, Vec<csp::Violation>) {
```

Comment:
As commented above, return only a `bool` here so the return-statements become easier and the violations are handled at the bottom of the method.